src/queue.py

```diff
@@ -17,18 +17,17 @@ def enqueue(self, value):
         """Add value to the top of  the queue."""
         return self._dlist.push(value)
 
-    def dequeue(self, value):
+    def dequeue(self):
```

Comment:
Your Queue.dequeue and therefore your your DList.shift methods should raise the appropriate Python exception.